actionpack/lib/action_controller/metal/instrumentation_payload.rb

```diff
@@ -0,0 +1,23 @@
+# frozen_string_literal: true
+
+module ActionController
+  # Encasulates logic needed to contruct payload for controller events.
+  module InstrumentationPayload
+    private
+      # The payload data used by the following instrumented events:
+      # * start_processing.action_controller
+      # * process_action.action_controller
+      def controller_instrumentation_payload
```

Comment:
I don't think this duplication is worth extracting another module. Can you revert this change and let the duplication exist? Also I don't think `unpermitted_parameters.action_controller` event needs the headers or the format for example, so the duplication will not be so big.